test/org/sagebionetworks/bridge/services/AuthenticationServiceImplTest.java

```diff
@@ -201,6 +225,63 @@ public void testSignOutWhenSignedOut() {
         assertNull(cacheProvider.getUserSessionByUserId(userId));
     }
     
+    @Test
+    public void signUpWillCreateDataGroups() {
+        authService = spy(authService);
+        optionsService = spy(optionsService);
+        authService.setOptionsService(optionsService);
+        
+        Study tempStudy = TestUtils.getValidStudy(AuthenticationServiceImplTest.class);
+        tempStudy.setDataGroups(Sets.newHashSet("group1","group2","group3"));
+        tempStudy = studyService.createStudy(tempStudy);
+
+        Set<String> list = Sets.newHashSet("group1");
+        String name = TestUtils.randomName(AuthenticationServiceImplTest.class);
+        String email = "bridge-testing+"+name+"@sagebase.org";
+        
+        try {
+            SignUp signUp = new SignUp(name, email, "P@ssword1", null, list);
+
+            authService.signUp(tempStudy, signUp, true);
+            Account account = accountDao.getAccount(tempStudy, email);
+            HealthId healthId = healthCodeService.getMapping(account.getHealthId());
+            
+            verify(authService).signUp(tempStudy, signUp, true);
+            verify(optionsService).setDataGroups(tempStudy, healthId.getCode(), new DataGroups(list));
+        } finally {
+            accountDao.deleteAccount(tempStudy, email);
+            studyService.deleteStudy(tempStudy.getIdentifier());
+        }
+    }
+    
+    @Test
+    public void userCreatedWithDataGroupsHasThemOnSignIn() throws Exception {
+        // @Before method always ensures there is one data group in the study. Use that for this test.
+        String group = studyService.getStudy(TestConstants.TEST_STUDY_IDENTIFIER).getDataGroups().iterator().next();
+        Set<String> dataGroups = Sets.newHashSet(group);
```

Comment:
Yes this seems convoluted as in the @Before we add a specific known group. Changed